### PR TITLE
[WFTC-69] FileSystemXAResourceRegistry: require permission to transac…

### PR DESCRIPTION
**Pull request by tadamski**
…tion log directory"

https://issues.jboss.org/browse/WFTC-69

@chengfang @fl4via 
Could you please review.

### CODE REVIEW

**Comment by chengfang** [comment]
While this makes for more fine-grained permission, it will require user/admin to know the path to xa recovery dir when designing security policy.  I guess many will simply use <<ALL FILES>> anyway for internal library code source.

**Comment by jamezp** [comment]
The path should be fairly well known so something like the following should work:
```
grant {
    permission java.io.FilePermission
        "/opt/wildfly/standalone/data/ejb-xa-recovery", "read-write";
    permission java.io.FilePermission
        "/opt/wildfly/standalone/data/ejb-xa-recovery/-", "read-write";
};
```

This seems somewhat related to https://issues.jboss.org/browse/WFLY-12171. There is a chat transcript with David linked in the JIRA.

**Comment by chengfang** [comment]
Okay, I see.  @jamezp thanks for explaining.

File: src/main/java/org/wildfly/transaction/client/provider/jboss/FileSystemXAResourceRegistry.java [hunk 1]
```diff
@@ -214,12 +213,12 @@ private void recoverInDoubtRegistries() throws IOException {
...
             this.filePath = xaRecoveryPath.resolve(xidString);
-            
+
             try {
                 fileChannel = doPrivileged((PrivilegedExceptionAction<FileChannel>) () -> {
                     final SecurityManager sm = System.getSecurityManager();
-                    if(sm != null) {
-                        sm.checkPermission(FILE_PERMISSION);
+                    if (sm != null) {
+                        sm.checkPermission(new FilePermission(xaRecoveryPath.toString() + File.separatorChar + '*', "read,write"));
```
Comment (bstansberry) [inline src/main/java/org/wildfly/transaction/client/provider/jboss/FileSystemXAResourceRegistry.java]: @tadamski If you cache this FilePermission in the containing FileSystemXAResourceRecoveryRegistry object you will save a per-transaction objection creation/gc.

**Comment by tadamski** [comment]
I have updated the PR so that the permission is cached.

**Pull request merged**